components/pipedrive/actions/add-activity/add-activity.mjs

```diff
@@ -1,133 +1,154 @@
-// legacy_hash_id: a_k6iKYA
-import { axios } from "@pipedream/platform";
+import pipedriveApp from "../../pipedrive.app.mjs";
+import utils from "../../common/utils.mjs";
 
 export default {
   key: "pipedrive-add-activity",
   name: "Add Activity",
-  description: "Adds a new activity. Includes more_activities_scheduled_in_context property in response's additional_data which indicates whether there are more undone activities scheduled with the same deal, person or organization (depending on the supplied data).",
-  version: "0.1.1",
+  description: "Adds a new activity. Includes more_activities_scheduled_in_context property in response's additional_data which indicates whether there are more undone activities scheduled with the same deal, person or organization (depending on the supplied data). See the Pipedrive API docs for Activities [here](https://developers.pipedrive.com/docs/api/v1/#!/Activities). For info on [adding an activity in Pipedrive](https://pipedrive.readme.io/docs/adding-an-activity)",
+  version: "0.1.3",
   type: "action",
   props: {
-    pipedrive: {
-      type: "app",
-      app: "pipedrive",
-    },
-    companydomain: {
-      type: "string",
-      description: "Your company name as registered in Pipedrive, which becomes part of Pipedrive API base url.",
-    },
+    pipedriveApp,
     subject: {
       type: "string",
+      label: "Subject",
       description: "Subject of the activity",
     },
     done: {
       type: "integer",
+      label: "Done",
       description: "Whether the activity is done or not. 0 = Not done, 1 = Done",
       optional: true,
     },
     type: {
       type: "string",
+      label: "Type",
       description: "Type of the activity. This is in correlation with the key_string parameter of ActivityTypes.",
     },
-    due_date: {
+    dueDate: {
       type: "string",
+      label: "Due Date",
       description: "Due date of the activity. Format: YYYY-MM-DD",
       optional: true,
     },
-    due_time: {
+    dueTime: {
       type: "string",
+      label: "Due Time",
       description: "Due time of the activity in UTC. Format: HH:MM",
       optional: true,
     },
     duration: {
       type: "string",
+      label: "Duration",
       description: "Duration of the activity. Format: HH:MM",
       optional: true,
     },
-    user_id: {
-      type: "integer",
+    userId: {
+      propDefinition: [
+        pipedriveApp,
+        "userId",
+      ],
```

Comment:
`userId` should use async options to list the available users.
https://developers.pipedrive.com/docs/api/v1/Users#getUsers